### ISSUE TITLE
[FEAT] Fan Control

### ISSUE THREAD

**Issue by HotBlack77** [AnalogJ/scrutiny#246]
**Is your feature request related to a problem? Please describe.**
I mainly use scrutiny to monitor drive health and in particular the keeping temperature of drives within a reasonable range is a key factor in keeping drives healthy. Also like most users, I am interested in keeping drives healthy, and cool, in a server, specifically a NAS, where you have several hard drives with one or several fans close to them, and do not want to let the motherboard control the fan speed, because it does so either statically, or using a temperature sensor unrelated to the real drive temperature (either on the CPU or on some other place on the motherboard).
By extending Scrutiny to also manage fans, Scrutiny would not just be a drive health monitoring tool, but also become able to actively help maintain drive health. 

**Describe the solution you'd like**
I would like to see Scrutiny include support though its GUI for setting the configuration options of HDD Fan control. HDD Fan control is a command line tool to dynamically control fan speed according to hard drive temperature on Linux (see more here: https://github.com/desbma/hddfancontrol ). HDD Fan control, can be run as a systemd service within the docker scrutiny docker container where the configuration is controlled through a simple configuration file. The idea would be that controls in the GUI of scrutiny lets you set parameters for your drives , which are then saved to the configuration file and when saved the service is then restarted to load the new configuration.
In addition pulling data on the fans and their current state and displaying it in the GUI would also be very nice.

**Additional context**
There already exists a docker container for running HDD Fan Control: https://github.com/fightforlife/docker_hddfancontrol 
It would be pretty straight forward to include the key elements of that docker compose file, and the script for running the service (its a very simple container) in scrutiny. Adding the GUI would be more complicated, but given how impressive the scrutiny GUI is, I suspect this projects maintainer would be well suited to the job.

**Comment by teambvd** [comment]
At a high level, I too like this idea... I've even tried my hand at addressing this myself, albeit via command line and cron. The problem unfortunately is that unlike with S.M.A.R.T., where we have a single tool that covers >99% of drives (smartctl), fan control is a whole other beast.

Some examples - 

- For supermicro [X9 and X10 boards](https://github.com/nimoatwoodway/freenas-ipmi-hdd-fan-control) (We're up to the X12/M12 generation currently) - this contains a wealth of other links with miscellaneous workarounds, and requires per-system modifications in many cases anyway)
- [Dell PowerEdge servers](https://github.com/tigerblue77/Dell_iDRAC_fan_controller_Docker) (but only those 13th gen and older, and for 13th gen, only if they've not updated to later iDRAC versions which unfortunately [strip away the ability to control the fans](https://www.dell.com/community/PowerEdge-Hardware-General/Dell-ENG-is-taking-away-fan-speed-control-away-from-users-iDrac/td-p/7441702))
- [ASRock RACK boards](https://www.asrockrack.com/support/faq.asp?id=38) could probably use similar methodology to the supermicros above, but with modified hex values for IPMItool
- Pretty much any Cisco hardware, which will basically tell you to [shove right off](https://community.cisco.com/t5/unified-computing-system/fan-policy-override/td-p/3787661)

With enough time and effort, nearly anything is possible... However with fan control being the absolute mess it currently is (as far as manufacturer implementation goes), I ended up keeping the two functions (drive health/history, and temperature defined fan monitoring/adjustment) separate. Probably be easy enough to integrate for most consumer boards as they've no lights-out management system to worry about, but for anything in the server space? Much more painful 😞

**Comment by HotBlack77** [comment]
I completely agree that fan control is much more of a mess than disk health monitoring. I also don't have much idea just how large a percentage of Scrutiny's users are using hardware which is incompatible with fan control, but for me it would work with my hardware (an older Asus intel motherboard).
The key point for me is having some means to do something about drive temps in the same system that actually monitors drive health, if you can control fans in the same system that has extensive data about drive health there should be good opportunities to build some logic that manages fans in a clever way in response to temperature changes.

**Comment by kaysond** [comment]
This issue is being closed automatically as a part of an effort to streamline and prioritize development effort. If it is still relevant, please reply to this comment and we will convert it to a discussion.
